### PR TITLE
Update semantic and resource attributes for spec v0.15.0

### PR DESCRIPTION
**Pull request by jkwatson**

### CODE REVIEW

File: semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java [hunk 1]
```diff
@@ -16,9 +16,696 @@
...
+  public static final AttributeKey<String> EXCEPTION_MESSAGE = stringKey("exception.message");
+
+  /**
+   * A stacktrace as a string in the natural representation for the language runtime. The
+   * representation is to be determined and documented by each language SIG.
+   */
+  public static final AttributeKey<String> EXCEPTION_STACKTRACE = stringKey("exception.stacktrace");
+
+  /** The name identifies the event. */
+  public static final AttributeKey<String> EVENT_NAME = stringKey("event.name");
```
Comment (jkwatson) [inline semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java:40]: I'm starting to wonder if maybe we should split the files out by type...i.e. event/span/metric

Comment (jack-berg) [reply semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java:40]: Maybe... the counter argument is that attributes may overlap between signals and are expected to have the same semantics. I.e. `MetricAttributes.HTTP_METHOD`, `SpanAttributes.HTTP_METHOD`.

Comment (jack-berg) [reply semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java:40]: What if we could annotate the attributes with some kind of indication on which signals they should be used with?

File: semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java [hunk 2]
```diff
@@ -16,9 +16,696 @@
...
+   */
+  public static final AttributeKey<String> PROCESS_EXECUTABLE_PATH =
+      stringKey("process.executable.path");
+
+  /**
+   * The command used to launch the process (i.e. the command name). On Linux based systems, can be
+   * set to the zeroth string in {@code proc/[pid]/cmdline}. On Windows, can be set to the first
+   * parameter extracted from {@code GetCommandLineW}.
+   */
+  public static final AttributeKey<String> PROCESS_COMMAND = stringKey("process.command");
```
Comment (jack-berg) [inline semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java:579]: Something doesn't seem right. There's lots of resource attributes in here.

File: semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java (hunk 1 above)
Comment (jkwatson) [reply semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java:40]: something to think about. I don't want to jump in any direction until things stabilize a bit more, though

File: semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java (hunk 2 above)
Comment (jkwatson) [reply semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java:579]: hmmm. the generator is not in a great state right now. probably something more needs tweaking. I tried only generating span attributes, but then we ended up with removed attributes that are tagged as type=event. Not sure how to properly tweak the generator at this point, to get what we want, and not get what we don't want.

File: semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java [hunk 3]
```diff
@@ -579,8 +681,18 @@ public final class SemanticAttributes {
    */
   public static final AttributeKey<String> HTTP_URL = stringKey("http.url");
 
-  /** The ordinal number of request re-sending attempt. */
-  public static final AttributeKey<Long> HTTP_RETRY_COUNT = longKey("http.retry_count");
```
Comment (jack-berg) [inline semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java:583]: We decided to not delete, and instead `@deprecate` correct?

File: semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java [hunk 4]
```diff
@@ -16,9 +16,139 @@
...
+   */
+  public static final AttributeKey<String> EVENT_DOMAIN = stringKey("event.domain");
+
+  /**
+   * The Amazon Resource Name (ARN) of an <a
+   * href="https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ECS_instances.html">ECS
+   * container instance</a>.
+   */
+  public static final AttributeKey<String> AWS_ECS_CONTAINER_ARN =
+      stringKey("aws.ecs.container.arn");
```
Comment (jack-berg) [inline semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java:62]: Something's still not right with the script. These are defined in the [ecs resource semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/94c9c75c4f82fbda08bddff02ce9a0c582fdd55c/specification/resource/semantic_conventions/cloud_provider/aws/ecs.md).

File: semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java [hunk 5]
```diff
@@ -949,7 +1078,50 @@ public final class SemanticAttributes {
...
+    public static final String BROWSER = "browser";
+    /** Events from mobile apps. */
+    public static final String DEVICE = "device";
+    /** Events from Kubernetes. */
+    public static final String K8S = "k8s";
+
+    private EventDomainValues() {}
+  }
+
+  public static final class AwsEcsLaunchtypeValues {
```
Comment (jack-berg) [inline semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java:1116]: Another instance of resource semantic conventions.

**Comment by jkwatson** [comment]
let's hold off until we can figure out how to get the script to do what we want.

File: semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java (hunk 3 above)
Comment (jkwatson) [reply semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java:583]: fixed

**Review by jack-berg — APPROVED** [review]

**Pull request merged**